### PR TITLE
Bayesian inference updates in manifest.toml and project.toml

### PR DESCRIPTION
**Pull request by ParamThakkar123**
## Checklist

- [X] Appropriate tests were added
- [X] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [X] Any new documentation only uses public API

### CODE REVIEW

**Comment by ParamThakkar123** [comment]
@ChrisRackauckas I was working on updating the PINN benchmarks but they always show the following error while updating the dependencies.

image

This occurs while updating almost any dependency. What can be done for this ??  Also I didn't find the repositories for OptimizationOptimJL, OptimizationFlux

**Comment by ParamThakkar123** [comment]
@ErikQQY Can you please take a look at this PR as well as the error that is repeatedly occurring here

**Comment by ErikQQY** [comment]
The compats you bumped have version conflicts, you would have to read the error message carefully and find the appropriate versions, you can refer to https://pkgdocs.julialang.org/dev/managing-packages/#conflicts-1 and https://discourse.julialang.org/t/a-guide-how-to-handle-error-unsatisfiable-requirements-detected/43406

**Comment by ParamThakkar123** [comment]
> The compats you bumped have version conflicts, you would have to read the error message carefully and find the appropriate versions, you can refer to https://pkgdocs.julialang.org/dev/managing-packages/#conflicts-1 and https://discourse.julialang.org/t/a-guide-how-to-handle-error-unsatisfiable-requirements-detected/43406

Got it. I'll go through this and fix things. Thank you.

**Comment by ParamThakkar123** [comment]
@ChrisRackauckas @ErikQQY Can you please take a look at this ?? I think this is done. All the benchmarks for BayesianInference including the manifest.toml has been updated and bumped.

**Comment by ChrisRackauckas** [comment]
https://buildkite.com/julialang/scimlbenchmarks-dot-jl/builds/2894#0192c1d7-e872-4975-8ff3-79c0898f56f0/561-678

you need to resolve on v1.10

**Comment by ParamThakkar123** [comment]
@ChrisRackauckas Can you please review this PR ?? I think this is done all the jmd files run without any errors, also the manifest and project.toml files are bumped to the latest versions on julia version 1.9.1

**Comment by ParamThakkar123** [comment]
DynamicalODE has been resolved on Julia version 1.10.4

File: benchmarks/BayesianInference/Project.toml [hunk 1]
```diff
@@ -15,16 +15,16 @@ StaticArrays = "90137ffa-7385-5640-81b9-e52037218182"
...
+Distributions = "0.25, 0.25.112"
+DynamicHMC = "3.1, 3.4.7"
+OrdinaryDiffEq = "6, 6.89.0"
+ParameterizedFunctions = "5.3, 5.17.0"
+Plots = "1.4, 2.0.0"
+RecursiveArrayTools = "2.5, 3.27.1"
+SciMLBenchmarks = "0.1, 0.1.3"
+StanSample = "7, 7.10.1"
+StaticArrays = "1, 1.9.8"
+Turing = "0.28, 0.30, 0.34.1, 0.35.1"
```
Comment (ChrisRackauckas) [inline benchmarks/BayesianInference/Project.toml:30]: Remove the lower ones and bump it?

Comment (ParamThakkar123) [reply benchmarks/BayesianInference/Project.toml:30]: @ChrisRackauckas I removed the lower versions and tried to bump them but I am getting an unsatisfiable requirements error for almost every module.

Comment (ParamThakkar123) [reply benchmarks/BayesianInference/Project.toml:30]: > Remove the lower ones and bump it?

 I removed the lower versions and tried to bump them but I am getting an unsatisfiable requirements error for almost every module after removing older versions

Comment (ChrisRackauckas) [reply benchmarks/BayesianInference/Project.toml:30]: Update to https://github.com/JuliaRegistries/General/pull/118430

Comment (ParamThakkar123) [reply benchmarks/BayesianInference/Project.toml]: @ChrisRackauckas Done

File: benchmarks/BayesianInference/Project.toml [hunk 2]
```diff
@@ -15,16 +15,16 @@ StaticArrays = "90137ffa-7385-5640-81b9-e52037218182"
...
```
Comment (ChrisRackauckas) [inline benchmarks/BayesianInference/Project.toml:30]: Only keep the top ones

**Comment by ChrisRackauckas** [comment]
Needs to bump the manifest

and remove the DynamicalODE stuff from this PR

**Comment by ParamThakkar123** [comment]
@ChrisRackauckas While keeping only the topic ones in the compat, the package errors with unsatisfiable requirements error for every package. How that should be dealt with ??

**Comment by ChrisRackauckas** [comment]
find the package that needs to bump

**Comment by ParamThakkar123** [comment]
@ChrisRackauckas The required changes have been made to the `Project.toml` file. `Manifest.toml` has been bumped and `DynamicalODE` has been removed from this PR

File: benchmarks/BayesianInference/Project.toml [hunk 3]
```diff
@@ -16,15 +16,15 @@ Turing = "fce5fe82-541a-59a6-adf8-730c64b5f9a0"
 
 [compat]
 BenchmarkTools = "1.0"
-DiffEqBayes = "3.0.1"
-DiffEqCallbacks = "2"
+DiffEqBayes = "3.2.1"
+DiffEqCallbacks = "2, 4.0.0"
```
Comment (ChrisRackauckas) [inline benchmarks/BayesianInference/Project.toml:20]:
```suggestion
DiffEqCallbacks = "4.0.0"
```

File: benchmarks/BayesianInference/Project.toml [hunk 4]
```diff
@@ -16,15 +16,15 @@ Turing = "fce5fe82-541a-59a6-adf8-730c64b5f9a0"
...
 Distributions = "0.25"
 DynamicHMC = "3.1"
 OrdinaryDiffEq = "6"
 ParameterizedFunctions = "5.3"
 Plots = "1.4"
-RecursiveArrayTools = "2.5"
+RecursiveArrayTools = "2.5, 3.27.1"
```
Comment (ChrisRackauckas) [inline benchmarks/BayesianInference/Project.toml:26]:
```suggestion
RecursiveArrayTools = "3.27.1"
```

File: benchmarks/BayesianInference/Project.toml [hunk 5]
```diff
@@ -16,15 +16,15 @@ Turing = "fce5fe82-541a-59a6-adf8-730c64b5f9a0"
...
 SciMLBenchmarks = "0.1"
 StanSample = "7"
 StaticArrays = "1"
-Turing = "0.28, 0.30"
+Turing = "0.28, 0.30, 0.34.1, 0.35.1"
```
Comment (ChrisRackauckas) [inline benchmarks/BayesianInference/Project.toml:30]:
```suggestion
Turing = "0.35.1"
```

**Comment by ChrisRackauckas** [comment]
It's not all on the latest majors.

**Comment by ParamThakkar123** [comment]
@ChrisRackauckas all changes so far have been made. Any more changes needed ??

**Comment by ErikQQY** [comment]
Seems Stan.jl backend still have error

**Comment by ParamThakkar123** [comment]
@ErikQQY What does the stan_inference error exactly point to ?? I went through the documentation and test files and made changes, but that didn't solve the error

**Comment by ErikQQY** [comment]
Are you sure `stan_inference(prob, t, data, priors; ......)` you are using is the correct syntax?

Quote from DiffEqBayes docs:
> stan_inference(prob::DiffEqBase.DEProblem, alg, t, data, priors = nothing;
    stanmodel = nothing, likelihood = Normal, vars = (StanODEData(), InverseGamma(3, 3)), sample_u0 = false, solve_kwargs = Dict(), diffeq_string = nothing, sample_kwargs = Dict(), output_format = :mcmcchains, print_summary = true, tmpdir = mktempdir())

**Comment by ChrisRackauckas** [comment]
What do you get locally?

**Comment by ParamThakkar123** [comment]
> What do you get locally?

Locally it gives an IOError while on the CI it shows some issues with stan_inference. Here's the issue: 

```
Warning: build_function(::Array{<:Equation}...) is deprecated. Use build_function(::AbstractArray...) instead.
└ @ Symbolics [C:\Users\Hp\.julia\packages\Symbolics\5BibL\src\build_function.jl:832](file:///C:/Users/Hp/.julia/packages/Symbolics/5BibL/src/build_function.jl:832)
┌ Info: [C:\Users\Hp\AppData\Local\Temp\jl_mR6Rav\parameter_estimation_model.stan](file:///C:/Users/Hp/AppData/Local/Temp/jl_mR6Rav/parameter_estimation_model.stan) updated.
└ @ StanBase [C:\Users\Hp\.julia\packages\StanBase\IgV1g\src\common\update_model_file.jl:34](file:///C:/Users/Hp/.julia/packages/StanBase/IgV1g/src/common/update_model_file.jl:34)
IOError: cd(""): invalid argument (EINVAL)

Stacktrace:
 [1] uv_error
   @ .\libuv.jl:100 [inlined]
 [2] cd(dir::String)
   @ Base.Filesystem .\file.jl:91
 [3] cd(f::StanSample.var"#1#2"{IOBuffer, String, String}, dir::String)
   @ Base.Filesystem .\file.jl:100
 [4] SampleModel(name::String, model::String, tmpdir::String)
   @ StanSample C:\Users\Hp\.julia\packages\StanSample\mnc5W\src\stanmodel\SampleModel.jl:122
 [5] stan_inference(prob::ODEProblem{Vector{Float64}, Tuple{Float64, Float64}, true, Vector{Float64}, FitzhughNagumo{var"###ParameterizedDiffEqFunction#293", var"###ParameterizedTGradFunction#294", var"###ParameterizedJacobianFunction#295", Nothing, Nothing, ODESystem}, Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}, SciMLBase.StandardODEProblem}, t::Vector{Float64}, data::Matrix{Float64}, priors::Vector{Truncated{Normal{Float64}, Continuous, Float64, Float64, Float64}}, stanmodel::Nothing; likelihood::Type, vars::Tuple{DiffEqBayes.StanODEData, InverseGamma{Float64}}, sample_u0::Bool, save_idxs::Nothing, diffeq_string::Nothing, alg::Symbol, reltol::Float64, abstol::Float64, maxiter::Int64, num_samples::Int64, num_warmups::Int64, num_cpp_chains::Int64, num_chains::Int64, num_threads::Int64, delta::Float64, output_format::Symbol, print_summary::Bool, tmpdir::String)
   @ DiffEqBayes C:\Users\Hp\.julia\packages\DiffEqBayes\gXPkn\src\stan_inference.jl:194
 [6] stan_inference
   @ C:\Users\Hp\.julia\packages\DiffEqBayes\gXPkn\src\stan_inference.jl:54 [inlined]
 [7] top-level scope
   @ .\timing.jl:273 [inlined]
 [8] top-level scope
   @ c:\Users\Hp\Desktop\SciMLBenchmarks.jl\benchmarks\BayesianInference\jl_notebook_cell_df34fa98e69747e1a8f8a730347b8e2f_X20sZmlsZQ==.jl:0
```

This is the only method that error after bumping updates of packages. rest seem to work fine without making updates to the existing code

**Pull request closed without merging**